kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemProperties.java

```diff
@@ -60,4 +60,12 @@
    * be closed and finalized once they reach this age.
    */
   public static final String ROLL_INTERVAL_S_PROP = "kite.writer.roll-interval-seconds";
+
+  /**
+   * Until HADOOP-9565 is available and fully adopted, need to make this configurable so that
+   * we can avoid multiple expensive copy operations when writing output to a file system that
+   * does not support fast file renaming. Default is false unless the FileSystem URI scheme is
+   * s3n or s3a.
+   */
+  public static final String OBJECTSTORE_FILESYSTEM = "kite.writer.object-store";
```

Comment:
Just one suggestion here: it seems like this decision should be based on whether the filesystem supports efficient rename operations, since future object stores may support rename (and s3 may get it some day). Perhaps this setting should be "kite.writer.supports-rename", and by default we assume that s3a and s3n filesystems do not?

I think this keeps Kite focused on the properties it's interested in...where it shouldn't really care or make assumptions about the properties of an object store.